src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -2310,6 +2310,26 @@ void processContextElement
                     getEntitiesCollectionName(tenant).c_str(),
                     query.toString().c_str()));
 
+    std::map<std::string, std::string>::iterator it = uriParams.find("op");
+
+    // if is an append and uri has not possibility of options
+    if (strcasecmp(action.c_str(), "append") == 0 && it == uriParams.end())
+    {
+        EntityIdVector vec;
+        vec.push_back(enP);
+
+        AttributeList attList;
+        Restriction res;
+        ContextElementResponseVector rsp;
+        std::string err;
+        entitiesQuery(vec, attList, res, &rsp, &err, true, tenant, servicePathV);
```

Comment:
entityQuery() is a "heavy" operation at it involves an extra query to the DB each time an entity is going to be created. It would be great to detect the situation at DB insert time, i.e. if the the document to be inserted already exists then return the error at that point. Maybe Mongo operations parameter allow to do something like that... I'm not sure right now.